applications/callflow/src/module/cf_branch_property.erl

```diff
@@ -0,0 +1,92 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2013-2016, 2600Hz INC
+%%% @doc
+%%%
+%%% "data":{
+%%%   "action":"branch", % the only one action supported currently
+%%%   "scope": "[device|user|account]",
+%%%   "property": "whatever_field_you_want"
+%%% }
+%%%
+%%% Branch to child whose name pointed in callflow's data section
+%%%
+%%% @end
+%%% @contributors
+%%%   SIPLABS LLC (Maksim Krzhemenevskiy)
+%%%-------------------------------------------------------------------
+-module(cf_branch_property).
+
+-include("../callflow.hrl").
+
+%% API
+-export([handle/2]).
+
+-spec handle(wh_json:object(), whapps_call:call()) -> 'ok'.
+handle(Data, Call) ->
+    Scope = wh_json:get_binary_value(<<"scope">>, Data),
+    PointedField = wh_json:get_binary_value(<<"property">>, Data),
+    ChildName = case is_binary(PointedField) of
```

Comment:
I would prefer you call a new function with `PointedField` and pattern match, versus all of these `is_*` case clauses.

```erlang
ChildName = find_child_name(PointedField),
...


find_child_name(Data, Call, 'undefined') -> 'undefined';
find_child_name(Data, Call, PointedField) ->
    Scope = wh_json:get_binary_value(<<"scope">>, Data),
    lookup_child_name(Scope, Call, PointedField).
```

Then call another well-named function to pattern match the `ChildName` returned.